File: BaseBall/BaseBallTest/domain/OpponentTest.swift
//
//  OpponentTest.swift
//  BaseBallTest
//
//  Created by KimMinSeok on 4/4/24.
//

import XCTest

final class OpponentTest: XCTestCase {

    func test_상대방은_숫자를_가지고_있다(){
        let opponent = Opponent(numberGenerator: RandomNumberGenerator())
        opponent.number = [1,2,3]
        
        XCTAssertEqual(opponent.number, [1,2,3])
    }

    func test_상대방은_게임을_시작할때_랜덤한_숫자를_생성한다(){
        let opponent = Opponent(numberGenerator: RandomNumberGenerator())
        
        XCTAssertNotEqual(opponent.number, [4,5,6])
    }

}

File: src/test/kotlin/baseball/domain/OpponentTest.kt
package baseball

import baseball.domain.Opponent
import baseball.domain.numbergenerator.NumberGenerator
import baseball.domain.numbergenerator.RandomNumberGenerator
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows

class OpponentTest {
    inner class FakeNumberGenerator : NumberGenerator {
        override fun get(): String = "123"
    }
    @Test
    fun `게임에서 사용할 숫자를 가진다`() {
        val opponent = Opponent(FakeNumberGenerator());
        assertThat(opponent.number).isEqualTo("123")
    }
}